rust_src/src/process.rs

```diff
@@ -556,6 +556,77 @@ pub fn internal_default_interrupt_process(
 #[rustfmt::skip]
 def_lisp_sym!(Qinternal_default_interrupt_process, "internal-default-interrupt-process");
 
+/// Kill process PROCESS. May be process or name of one.
+/// See function `interrupt-process' for more details on usage.
+#[lisp_fn(min = "0")]
+pub fn kill_process(process: LispObject, current_group: LispObject) -> LispObject {
+    unsafe {
+        process_send_signal(process, libc::SIGKILL, current_group, false);
+    }
+    process
+}
+
+/// Send QUIT signal to process PROCESS. May be process or name of one.
+/// See function `interrupt-process' for more details on usage.
+#[lisp_fn(min = "0")]
+pub fn quit_process(process: LispObject, current_group: LispObject) -> LispObject {
+    unsafe {
+        process_send_signal(process, libc::SIGQUIT, current_group, false);
+    }
+    process
+}
+
+/// Stop process PROCESS. May be process or name of one.
+/// See function `interrupt-process' for more details on usage.
+/// If PROCESS is a network or serial or pipe connection, inhibit handling
+/// of incoming traffic.
+#[lisp_fn(min = "0")]
+pub fn stop_process(process: LispObject, current_group: LispObject) -> LispObject {
+    let mut p_ref: LispProcessRef = get_process(process).into();
+    let process_type = p_ref.ptype();
+    if process_type.eq(Qnetwork) || process_type.eq(Qserial) || process_type.eq(Qpipe) {
+        unsafe {
+            delete_read_fd(p_ref.infd);
+            p_ref.command = Qt;
+        }
+        return process;
+    }
+
+    unsafe {
+        process_send_signal(process, libc::SIGTSTP, current_group, false);
+    }
+
+    process
+}
+
+/// Continue process PROCESS. May be process or name of one.
+/// See function `interrupt-process' for more details on usage.
+/// If PROCESS is a network or serial process, resume handling of incoming
+/// traffic.
+#[lisp_fn(min = "0")]
+pub fn continue_process(process: LispObject, current_group: LispObject) -> LispObject {
+    let mut p_ref: LispProcessRef = process.into();
+    let process_type = p_ref.ptype();
+    if process_type.eq(Qnetwork) || process_type.eq(Qserial) || process_type.eq(Qpipe) {
+        unsafe {
+            if p_ref.command.eq(Qt)
+                && p_ref.infd >= 0
+                && (!p_ref.filter.eq(Qt) || p_ref.status.eq(Qlisten))
+            {
+                add_process_read_fd(p_ref.infd);
+                tcflush(p_ref.infd, libc::TCIFLUSH);
```

Comment:
What about Windows support?

Reply:
Really I have no any experience with Windows systems. I'll check their signals capabilities a bit later.

Reply:
Microsoft nows makes Win10 VirtualBox images available for download. You can pull it down, add in Rust and try to build.

Reply:
We used to use `appveyor` to run a Windows check for us like Travis does for Linux and MacOS but it was very unreliable.